sopel-modules/rat-board.py

```diff
@@ -1820,34 +1854,97 @@ def cmd_nick(bot, trigger, case, newnick):
     save_case_later(bot, case, forceFull=True)
     bot.say('Set Nick to ' + str(newnick))
 
-@commands('quiet', 'lastsignal', 'last')
-@require_rat('Sorry, but you need to be a registered and drilled Rat to use this command.')
-def cmd_quiet(bot, trigger):
+
+def func_quiet(bot, trigger, platform=''):
     """
     Tells the time since the last Signal
     """
-    if bot.memory['ratbot']['lastsignal'] is None:
-        bot.say("Sadly, I don't remember when we had the last signal. Maybe it was 42 seconds ago?")
-        return
-    tdelta = datetime.datetime.utcnow() - bot.memory['ratbot']['lastsignal']
+    retVal = None
+    if trigger.is_privmsg: retVal = NOLIMIT
+    # returning NOLIMIT doenst trigger the rate-limit
+    # therefor PMs arent rate-limited
+    # any other returned value is ignored
+
+    retPlat = ''
+    if platform == 'xb' : retPlat = 'Xbox'
+    if platform == 'pc' : retPlat = 'PC'
+    if platform == 'ps' : retPlat = 'PS4'
+
+    if platform != '':
+       platform = '-' + platform
+
+    if bot.memory['ratbot']['lastsignal' + platform] is None:
+        bot.say("Sadly, I don't remember when we had the last {0} signal. Maybe it was 42 seconds ago?"
+                .format(retPlat))
+        return retVal
+    tdelta = datetime.datetime.utcnow().replace(tzinfo=None) - bot.memory['ratbot']['lastsignal' + platform].replace(tzinfo=None)
     seconds = tdelta.seconds % 60
     minutes = int(tdelta.seconds / 60)
     hashours = False
     hours = 0
     if minutes >= 60:
-        hours = int(minutes/60)
+        hours = int(minutes / 60)
         minutes = minutes % 60
         hashours = True
     ret = str(minutes) + " minutes and " + str(seconds) + " seconds"
     if hashours:
         ret = str(hours) + " hours, " + ret
 
+    if retPlat != '': retPlat = 'on ' + retPlat
     if hours > 12:
-        bot.say("Wow, the last signal was so long ago... " + ret + " ago to be exact! Orangey approves!")
-        return
-    bot.say("It has been quiet for " + ret + "! Time to summon a case?")
+        bot.say("Wow, the last signal {0} was so long ago... {1} ago to be exact! Orangey approves!"
+                .format(retPlat, ret))
+        return retVal
+
+    bot.say("It has been quiet {0} for {1} ! Time to summon a case?"
+            .format(retPlat, ret))
+    return retVal
+
+
+@commands('quiet', 'lastsignal', 'last')
+@rate(600, 60)
+# the same user may only trigger once every 10 minutes,
+# it may only be triggered once every minute per channel
+# not rate limited in PMs
+@require_rat('Sorry, but you need to be a registered and drilled Rat to use this command.')
+def cmd_quiet(bot, trigger):
+    return func_quiet(bot, trigger)
+
+
+@commands('quiet-xb', 'lastsignal-xb', 'last-xb')
```

Comment:
Honestly it seems wasteful to register 4 hooks for the same command, can these be integrated into one command hook?

Reply:
You could make the platform a parameter or add it to quiet in general. Output "formatting" isn't too hard to change.